numaprom/udf/preprocess.py

```diff
@@ -18,10 +18,16 @@ def preprocess(_: str, datum: Datum) -> bytes:
     payload = StreamPayload(**orjson.loads(_in_msg))
     _LOGGER.info("%s - Received Payload: %r ", payload.uuid, payload)
 
+    # Load config
+    metric_config = get_metric_config(
```

Comment:
skimming through the code, it looks like `get_metric_config` will always hit disk unless `OmegaConf` caches the loaded files and reread only if there are changes. at scale, this might cause issues.

Reply:
Agreed. Even a 10 min pod level caching should be fine, right?

Reply:
This changes once I implement the watchdog observer

Reply:
we will be loading the config only when the files are changed